File: android/src/main/java/com/reactnativelibrary/swisseph/SwissephModule.java
package com.reactnativelibrary.swisseph;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactModuleWithSpec;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.turbomodule.core.interfaces.TurboModule;

import java.util.Iterator;
import java.util.List;
import java.util.Map;
import java.util.HashMap;

@ReactModule(name = SwissephModule.NAME)
public class SwissephModule extends ReactContextBaseJavaModule implements ReactModuleWithSpec, TurboModule {

  public static final String NAME = "Swisseph";

  SwissephModule(ReactApplicationContext context) {
    super(context);
    SwissephBridge.createInstance(context);
  }


  @NonNull
  @Override
  public String getName() {
    return NAME;
  }

  public void initialize(){
    super.initialize();

  }

  public void invalidate() {
    SwissephBridge.getInstance().destroy();
    super.invalidate();
  }

}

File: android/src/main/java/com/swisseph/SwissephModule.kt
package com.swisseph

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = SwissephModule.NAME)
class SwissephModule(reactContext: ReactApplicationContext) :
  NativeSwissephSpec(reactContext) {

  override fun getName(): String {
    return NAME
  }

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  override fun multiply(a: Double, b: Double): Double {
    return a * b
  }

  companion object {
    const val NAME = "Swisseph"
  }
}
